static/js/theme-map/ThemeMapConfig.js

```diff
@@ -186,6 +186,12 @@ export default class ThemeMapConfig {
      */
     this.dragEndListener = jsonConfig.dragEndListener || function () {};
 
+    /**
+     * Callback for when a map pan event has finished
+     * @type {Function}
+     */
+    this.panHandler = jsonConfig.panHandler || function () {};
```

Comment:
What do you think of calling this `panEndListener` for consistency? It may also be a bit more descriptive as the listener is fired on the end of the pan event

Reply:
I left it as is because it was already named `panListener ` in Map.js (consulting's code). Also, it would be misleading because Google Maps fires repeatedly. We could make a new handler called `panEndListener`
which is the same as the `panHandler` for MapBox, and for Google Maps, we debounce it and call it when the map has settled. What do you think of that idea?

Reply:
Correct, that would mean that Google would would have both a `panHandler` and a `panEndListener`, however our code would only use the `panEndListener`

Reply:
I would feel like we want to only `updateMapPropertiesInStorage` on panEnd. What do you think of making that `panEnd` like you suggested, following the zoomend in Google's case?

Reply:
Sure, I can make a `panEndListener`, but what do you mean by "following the zoomend in Google's case"?

Reply:
The case where we simulate a zoomend by listening for idle once, after a zoom_changed event!

Reply:
The Google maps 'idle' event works correctly when dragging the map by mouse, but it fires repeatedly when the keyboard is used, which suggests that this is a Google maps issue. I have tried preventing calling the `panHandler` if the navigation keys are pressed, however it appears that Google's map captures the navigation key presses, and therefore I'm unable to detect if they are pressed from our javascript. So at this point, we have a. couple of options:
1. Move forward with this PR as-is
2. Add a simple time-based debouncer to help reduce the frequency of panHandler calls performed when the map is moving.
I also think we may be able to optimize the javascript on the main thread so that we don't drop frames every time a search is performed, but that's out of scope of this item. What are your thoughts?